### ISSUE TITLE
Client crash when using crafting upgrade in backpacks (java.lang.RuntimeException: Slot 13 not in valid range)

### ISSUE THREAD

**Issue by rckymtnrfc** [P3pp3rF1y/SophisticatedCore#236]
It doesn't happen all the time, but occasionally when crafting using the crafting upgrade in a backpack, the client will crash.  It seems to be very random.  I can play for hours without it happened and then suddenly the client crashes.  At first I thought I was just pulling items out too quickly but it's happen when pulling out a single item. 

I haven't tested with crafting upgrades in chests, fyi.

Thanks for any help you could provide with this. 

Forge 47.2.20
Soph Core 0.6.21.609
Soph Backpacks 3.20.5.1044
Soph Storage 0.10.21.793

Crash report - https://mclo.gs/YJDJ7uV

**Comment by P3pp3rF1y** [comment]
I have seen this one once or twice before. Looks like incorrect recipe that is trying to put items back into crafting grid after the craft (similar to buckets being put back in vaniila cake crafting recipe) outside of 3x3 grid. So this is basically issue of the mod that's adding it, but I am going to put in some code to prevent crash / log the offending recipe in the log.

**Comment by P3pp3rF1y** [comment]
Actually if you happen to find out which recipe is causing this that would be helpful to verify the theory / the defensive code that I want to include.

**Comment by rckymtnrfc** [comment]
Off hand, I don't recall what recipes caused it but when I get the crash again, I'll report back with the recipe I was crafting.  Thanks!

**Comment by rckymtnrfc** [comment]
Just got the crash.  I was crafting vanilla, Spruce trapdoors.  I put in 2 planks in each of the bottom 6 slots and then quickly pulled out the trap doors (didn't shift click) and it crashed.

**Comment by P3pp3rF1y** [comment]
Well, that looks like some mod is modifying that recipe or more probably modifies handling of stacks being set to slots. 

I am unable to recreate with just sophisticated and now I am not even sure that the "fix" I was planning for makes sense here as whatever it is is trying to set random slots and it's likely that it affects more than just crafting upgrade.

**Comment by P3pp3rF1y** [comment]
Actually I take that last comment back, it is still an issue with crafting recipe, but obviously not with vanilla one so something is either changing that or is modifying how that or probably more vanilla recipes work which is causing more remaining items to get returned by the recipe than can fit in the crafting grid so I am still going to add prevention of hard crash and will log errors with recipe id and what kind of items it is trying to put in what slots

**Comment by James103** [comment]
I just crashed in the [Project Architect 2](https://www.curseforge.com/minecraft/modpacks/project-architect-2) modpack, while crafting Wind Turbines from Mekanism using JEI to fill in the recipe before shift-clicking the output. I was able to craft several wind turbines through the same method without crashing; it was only on one of the repetitions that it crashed.
See https://github.com/ChosenArchitect/Project-Architect-2/issues/2#issuecomment-2254301492 for the crash log.
